flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -51,22 +58,39 @@ public class ExecutionFailureHandler {
     /** Number of all restarts happened since this job is submitted. */
     private long numberOfRestarts;
 
+    private final Context taskFailureCtx;
+    private final Context globalFailureCtx;
+    private final Collection<FailureEnricher> failureEnrichers;
+    private final ComponentMainThreadExecutor mainThreadExecutor;
+
     /**
      * Creates the handler to deal with task failures.
      *
      * @param schedulingTopology contains the topology info for failover
      * @param failoverStrategy helps to decide tasks to restart on task failures
      * @param restartBackoffTimeStrategy helps to decide whether to restart failed tasks and the
      *     restarting delay
+     * @param mainThreadExecutor the main thread executor of the job master
+     * @param failureEnrichers a collection of {@link FailureEnricher} that enrich failures
+     * @param taskFailureCtx Task failure Context used by FailureEnrichers
+     * @param globalFailureCtx Global failure Context used by FailureEnrichers
      */
     public ExecutionFailureHandler(
             final SchedulingTopology schedulingTopology,
             final FailoverStrategy failoverStrategy,
-            final RestartBackoffTimeStrategy restartBackoffTimeStrategy) {
+            final RestartBackoffTimeStrategy restartBackoffTimeStrategy,
+            final ComponentMainThreadExecutor mainThreadExecutor,
+            final Collection<FailureEnricher> failureEnrichers,
+            final Context taskFailureCtx,
+            final Context globalFailureCtx) {
 
         this.schedulingTopology = checkNotNull(schedulingTopology);
         this.failoverStrategy = checkNotNull(failoverStrategy);
         this.restartBackoffTimeStrategy = checkNotNull(restartBackoffTimeStrategy);
+        this.mainThreadExecutor = checkNotNull(mainThreadExecutor);
+        this.failureEnrichers = checkNotNull(failureEnrichers);
+        this.taskFailureCtx = taskFailureCtx;
```

Comment:
We actually never expect this context to be null in production. 
I see that it is not checked to be non-null because it can be `null` in tests. Usually we will introduce a `DummyFailureEnricherContext` for this purpose.
But since there is a follow-up task FLINK-32114, I think we can refine it there. (maybe by introducing `FailureEnricherContextFactory` and `DummyFailureEnricherContextFactory`?)

Reply:
Sure, update FLINK-32114 with your comment